<launch>
<include file="$(find kcctslam)/launch/sensor.launch"/>

<node pkg="turtlesim" type="turtle_teleop_key" name="turtle_teleop_key" output="screen"/>
<node name="nav_master" pkg="kcctslam" type="nav_master" output="screen">
	<param name="waypointfile" type="str" value="$(find kcctslam)/config/waypointdata/wpdata.csv" />
	</node>
    	<node pkg="kcctslam" name="laser2pc" type="laser2pc.py" output="screen">
	</node>


	<!-- 近傍点探索に使うパラメータと結果が入る変数
	radius: 半径 r [m].
	max_nn: 最近傍のお隣さんを何点見つかったら探索を打ち切るか。(nn = nearest neighbor)
		0にすると打ち切らない
	-->	
	
	<node pkg="kcctslam" name="pcl_handler" type="pcl_handler" >
	<param name="max_nn" value="20"/>
	<param name="radius" value="10.0"/>
	<param name="x_max" value="5.0"/>
	<param name="x_min" value="-0.5"/>
	<param name="y_max" value="0.4"/>
	<param name="y_min" value="-0.4"/>
	<param name="loop_rate" value="10"/>		
	</node>
</launch>